src/utils/CalendarUtils/CalendarUtils.js

```diff
@@ -37,30 +38,66 @@ export const elementExists = selector => {
   return javaScriptUtils.isDefined(element);
 };
 
-export const filterContributionDays = (dailyData, timeframe) => {
-  const isDay = dailyData.attributes.class === "day";
-
-  // Weekdays and months displayed around the calendar should be disregarded.
-  if (!isDay) {
-    return false;
-  }
-
+export const filterByTimeframe = (dailyDataWithContributions, timeframe) => {
   if (!timeframe.start && !timeframe.end) {
-    return true;
+    return dailyDataWithContributions;
   }
 
-  const contributionsDate = new Date(dailyData.attributes["data-date"]);
+  const filtered = Object.entries(dailyDataWithContributions).filter(
+    ([date]) => {
+      const convertedDate = new Date(date);
 
-  if (timeframe.start && !timeframe.end) {
-    return contributionsDate >= new Date(timeframe.start);
-  }
+      if (timeframe.start && !timeframe.end) {
+        return convertedDate >= new Date(timeframe.start);
+      }
 
-  if (!timeframe.start && timeframe.end) {
-    return contributionsDate <= new Date(timeframe.end);
-  }
+      if (!timeframe.start && timeframe.end) {
+        return convertedDate <= new Date(timeframe.end);
+      }
 
-  return (
-    contributionsDate >= new Date(timeframe.start) &&
-    contributionsDate <= new Date(timeframe.end)
+      return (
+        convertedDate >= new Date(timeframe.start) &&
+        convertedDate <= new Date(timeframe.end)
+      );
+    }
   );
+
+  return Object.fromEntries(filtered);
 };
+
+export const extractWeeks = calendar => {
```

Comment:
Would this make sense?

```suggestion
export const extractWeeksFromGitHub = calendar => {
```

Reply:
Hmmm, I think so, thanks!

Reply:
https://github.com/c-hive/team-contribution-calendar/pull/70/commits/e5d5689c479c020e62aff1575ffb758d0327ef80